### DISCUSSION TITLE
Unable to import static file (images) when using tamagui@latest

### DISCUSSION THREAD

**Discussion by akinwale-habibullah** [tamagui/tamagui#2461] [General]
### Current Behavior

On creating a new tamagui starter app using npm create tamagui@latest, which is currently 1.92.0, and selecting the `Free - Expo + Next in a production ready monorepo`.

I created an `assets` folder in the packages/app folder and drop some images there.

Currently, if I import the file like this:

> import imageName from '../../assets/sampleImage.png'

I get this error:

> Cannot find module '../../assets/imageName.png' or its corresponding type declarations.ts(2307)

### Expected Behavior

I dont expect any error. I expect to be able to use this image in an Image tag's source prop

### Tamagui Version

```markdown
1.92.0
```

### Platform (Web, iOS, Android)

```markdown
Web, IOS, Android
```

### Reproduction

```markdown
Create a new starter using npm create tamagui@latest, and select the `   Free - Expo + Next in a production ready monorepo` template

Create an assets folder in the packages/app folder, drop an image inside and try to import the image in features/home/home.tsx or any component you create
```

### System Info

```markdown
Cannot find module '../../assets/family-meal-1.jpg' or its corresponding type declarations.ts(2307)
```

**Comment by natew** [comment]
Migrated to discussion, this is something you have to set up right and we don't have docs yet, if you want to help document and figure out a good pattern that'd be useful.

**Reply by akinwale-habibullah** [reply]
Hi Nate, thanks for the timely feedback. How do I set it up right? and I wouldn't mind helping with the documentation. How do I proceed with this?

**Comment by akinwale-habibullah** [comment]
I resolved the issue by;

1. Adding file-loader dependency to the `packages/app` directory
2. Adding type declarations to types.d.ts in the `packages/app` directory

**Reply by kirgy** [reply]
@akinwale-habibullah do you have specific examples of what this looks like? It would realy help with documentation which I'd be happy to tackle.

**Reply by KROT47** [reply]
For me worked adding this to `packages/app/types.d.ts`:

```ts
declare global {
  type LocalImageSource =
    | {
        src: string
        width: number
        height: number
        blurDataURL: string
        blurWidth: number
        blurHeight: number
      }
    | number

  declare module '*.png' {
    const content: LocalImageSource
    export default content
  }

  declare module '*.jpg' {
    const content: LocalImageSource
    export default content
  }
}
```

`number` is for react native

**Reply by whittenator** [reply]
@KROT47 So after you add the above, how would one reference a .svg example in a React component? Can you please give an example?

**Reply by LoadJulz** [reply]
@whittenator Just declare your svg declaration. 

`declare module '*.svg' {
    const content: LocalImageSource
    export default content
  }`

But if you want to use local svgs, I would recommend to build jsx components that contain your svg instead of loading them from an asset folder. You can do this with _react-native-svg_

**Comment by LoadJulz** [comment]
Thanks for the solution provided by @akinwale-habibullah and @KROT47 

I think this can be useful for some more people. To load the image in the end I have build a AssetImage component which depending on react-native or next.js loads images properly:

```tsx
import React from 'react'
import { Image } from '@my/ui'

interface AssetImageProps {
  icon: number | { src: string }
}

const AssetImage: React.FC<AssetImageProps> = ({ icon }) => {
  if (typeof icon === 'number') {
    return <Image height={100} width={100} source={icon} />
  }
  return <Image height={100} width={100} src={icon.src} />
}

export default AssetImage
```

We just need to differentiate between the type number as React Native is handling local assets differently. I have tested this for .jpg .webp and .png. For SVGs it is not working on React Native, but I rather would recommend to use _react-native-svg_ for this case.
